pkg/cfn/builder/managed_launch_template.go

```diff
@@ -19,6 +19,23 @@ func (m *ManagedNodeGroupResourceSet) makeLaunchTemplateData(ctx context.Context
 		MetadataOptions:   makeMetadataOptions(mng.NodeGroupBase),
 	}
 
+	if mng.CapacityReservation != nil {
+		valueOrNil := func(value *string) *gfnt.Value {
+			if value != nil {
+				return gfnt.NewString(*value)
+			}
+			return nil
+		}
+		launchTemplateData.CapacityReservationSpecification = &gfnec2.LaunchTemplate_CapacityReservationSpecification{}
+		launchTemplateData.CapacityReservationSpecification.CapacityReservationPreference = valueOrNil(mng.CapacityReservation.CapacityReservationPreference)
+		if mng.CapacityReservation.CapacityReservationTarget != nil {
+			launchTemplateData.CapacityReservationSpecification.CapacityReservationTarget = &gfnec2.LaunchTemplate_CapacityReservationTarget{
+				CapacityReservationId:               valueOrNil(mng.CapacityReservation.CapacityReservationTarget.CapacityReservationID),
+				CapacityReservationResourceGroupArn: valueOrNil(mng.CapacityReservation.CapacityReservationTarget.CapacityReservationResourceGroupARN),
```

Comment:
The [CapacityReservationSpecification](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CapacityReservationSpecification.html) docs specify that:
```
You can specify only one parameter at a time. If you specify CapacityReservationPreference and CapacityReservationTarget, the request fails.
```
We should add validation that not both are set at the same time.